crates/compiler/builtins/roc/List.roc

```diff
@@ -694,19 +694,19 @@ range = \{ start, end, step ? 0 } ->
         At at ->
             isComplete =
                 if stepIsPositive then
-                    \i -> i > at
+                    \i -> i >= at
                 else
-                    \i -> i < at
+                    \i -> i <= at
 
             # TODO: switch to List.withCapacity
             rangeHelp [] inclusiveStart incByStep isComplete
 
         Before before ->
             isComplete =
                 if stepIsPositive then
-                    \i -> i >= before
+                    \i -> i >= before + 1
                 else
-                    \i -> i <= before
+                    \i -> i <= before - 1
```

Comment:
Could be `i < before` to save one arithmetic operation! 😃

Reply:
This is embarrassing but I actually just had these backwards :joy: